pkg/config/config_test.go

```diff
@@ -43,19 +80,14 @@ func TestNewFromFlagsKubeconfigEnv(t *testing.T) {
 	if config.Kubeconfig != testVal {
 		t.Errorf("kubeconfig option not loaded correctly from ebv variable, expected: %s, got: %s", testVal, config.Kubeconfig)
 	}
+	os.Unsetenv("KUBECONFIG")
 }
 
 func TestNewFromFlagsKubeconfigHome(t *testing.T) {
 	// reset for testing
 	pflag.CommandLine = pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
 
 	expected := homedir.HomeDir() + "/.kube/config"
-
-	err := os.Unsetenv("KUBECONFIG")
```

Comment:
test should be self contained, these leak like crazy.

Reply:
Exactly - the test expects the `KUBECONFIG` not to be set in the system.

If we won't unset it, it will be prone to failures based on the external state of the environment, i.e.:
- user A runs it, not having `KUBECONFIG` set -> test works
- user B has `KUBECONFIG` set on his system -> test will wail

Reply:
As to leaking - what kind of leak do you have in mind?